### PR TITLE
Convert InputStreamSupplier to lambdas

### PR DESCRIPTION
**Pull request by jorsol**
This updates the current code to use Java 8 lambdas when a `InputStreamSupplier` is used reducing some boilerplate code.

### CODE REVIEW

File: src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java [hunk 1]
```diff
@@ -377,9 +377,9 @@ private void writeManifest( ConcurrentJarCreator zOut, Manifest manifest )
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         manifest.write( baos );
 
-        ByteArrayInputStream bais = new ByteArrayInputStream( baos.toByteArray() );
-        super.zipFile( createInputStreamSupplier( bais ), zOut, MANIFEST_NAME, System.currentTimeMillis(), null,
-                       DEFAULT_FILE_MODE, null, false );
+        InputStreamSupplier bais = () -> new ByteArrayInputStream( baos.toByteArray() );
```
Comment (plamentotev) [inline src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java:380]: Unless I'm missing something in all cases where `createInputStreamSupplier`, is called it is called with byte array. Maybe we can keep the method and reduce the boilerplate even more by moving the `ByteArrayInputStream bais = new ByteArrayInputStream( ... )` part there.

File: src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipArchiver.java [hunk 2]
```diff
@@ -506,22 +508,15 @@ protected void zipFile( final ArchiveEntry entry, ConcurrentJarCreator zOut, Str
...
-                }
-                catch ( IOException e )
-                {
-                    throw new RuntimeException( e );
-                }
+                return entry.getInputStream();
+            }
+            catch ( IOException e )
+            {
+                throw new UncheckedIOException( e );
```
Comment (plamentotev) [inline src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipArchiver.java:518]: I didn't knew about this new exception. Looks really useful for lambdas. Thanks for sharing it.

**Review by plamentotev — APPROVED** [review]

File: src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java (hunk 1 above)
Comment (jorsol) [reply src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java]: Well, I have renamed it to `in` as is the parameter name for zipFile method, there are other inconsistencies in naming, as in other places it's called payload, and source.

File: src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java [hunk 3]
```diff
@@ -139,7 +139,7 @@
      * <p/>
      * Will not be filled unless the user has asked for an index.
      */
-    private Vector<String> rootEntries;
+    private ArrayList<String> rootEntries;
```
Comment (slachiewicz) [inline src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java:142]: List

File: src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java [hunk 4]
```diff
@@ -722,21 +723,9 @@ protected static String findJarName( String fileName, String[] classpath )
             return new File( fileName ).getName();
```
[the remark this answers is not in the document]
Comment (jorsol) [reply src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java:723]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (jorsol) [reply src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java:723]: @sonatype-lift ignore

**Review by slachiewicz — APPROVED** [review]

**Pull request merged**